src/bitwarden_workflow_linter/rules/name_capitalized.py

```diff
@@ -56,7 +56,7 @@ def fn(self, obj: Union[Workflow, Job, Step]) -> Tuple[bool, str]:
                 if obj.name[0] != "_":
                     return obj.name[0].isupper(), self.message
         else:
-            if obj.name:
+            if obj.name and "matrix" not in obj.name:
                 return obj.name[0].isupper(), self.message
```

Comment:
This looks to be trying to avoid incorrectly catching things like: `name: ${{ matrix.something }}"`
I think we still want to catch things like `name: lowercase text ${{ matrix.something}}"`.

So we can one of two things:
Allow variables as the start of names (including using matrix)
```suggestion
            if obj.name:
                return obj.name[0].isupper() or obj.name[0] == "$", self.message
```

OR
Do not allow using variables as the start of a name (This is my preference, but I'm not sure if it's compatible with existing workflows). This would require having a valid capital letter first and fail if symbols or lowercase letters are used as upper only returns true if it is a letter and is capital.
```suggestion
            if obj.name:
                return obj.name[0].isupper(), self.message
```

Reply:
Some jobs or steps in some workflows use matrix, so it's a valid part of our workflow standard. using `matrix` to catch a passing job or step name is more restrictive and appropriate to me because it's what we expect to be present in the workflows name. using `obj.name[0] == "$"` opens more room for error.

Reply:
The existing case we want to allow then is something like:
`name: ${{ matrix.project_name }}.zip`

Since `name: My Job - ${{ matrix.something }}` would still pass correctly

Right now the logic would also incorrectly pass
`name: lowercase word matrix`

What do you think about something along these lines to reduce it to only capitalized or `${{ matrix... }}...`
```suggestion
            if obj.name and not re.match(r"^\s*\${{\s*matrix\..*}}.*", obj.name) ):
                return obj.name[0].isupper(), self.message
```

Reply:
@pixman20 This is better. I can see the loophole in the example you referenced, `name: lowercase word matrix`.
The suggested regex has some bugs. I will test the regex locally and push a commit.